Fix handling of skipped binding constraints

flomnes opened the pull request:
Fix #355

flomnes committed:
Use pointer variable for study->uiinfo->constraint(x)

Hopefully, this will increase readibility.

flomnes committed:
Introduce and use BindingConstraint::skipped()

flomnes committed:
Add !skipped() condition to BindConstList::eachEnabled

flomnes commented on src/libs/antares/study/constraint/constraint.hxx:
```diff
@@ -114,6 +114,11 @@ inline BindingConstraint::const_iterator BindingConstraint::end() const
 
+inline bool BindingConstraint::skipped() const
+{
+    return !(linkCount() > 0 || enabledClusterCount() > 0);
```
`linkCount() == 0 && enabledClusterCount() == 0`

guilpier-code replied:
I do agree with your proposal of change

guilpier-code replied:
This comment to tell gitHub I have requested changes (the same as those spotted by flomnes)

flomnes reviewed (commented):
Add provided study to Antares_Simulator_Tests repository, run as end-to-end test.

flomnes commented on src/ui/simulator/toolbox/components/datagrid/renderer/bindingconstraint/offsets.cpp:
```diff
@@ -158,9 +160,8 @@ IRenderer::CellStyle LinkOffsets::cellStyle(int x, int y) const
-        return ((study->uiinfo->constraint(x))->enabled()
-                && ((study->uiinfo->constraint(x))->linkCount() > 0
-                    || (study->uiinfo->constraint(x))->enabledClusterCount() > 0))
+        return (constraint->enabled()
+                && (constraint->linkCount() > 0 || constraint->enabledClusterCount() > 0))
```
use `constraint->skipped()`

guilpier-code replied:
I do agree with your proposal of change

flomnes committed:
PR remarks

- Use constraint->skipped() where needed
- Use constraint instead of study->uiinfo->constraint(x)
- Simplify implementation of BindingConstraint::skipped
